### PR TITLE
MAINT: Enable `main` for Private Notebook Launcher Links

### PR DESCRIPTION
**Pull request by aadsouza**

### CODE REVIEW

**Comment by mmcky** [comment]
@aadsouza you may need to change

```
git+https://github.com/QuantEcon/quantecon-book-theme@jupyterlab
```

to 

```
git+https://github.com/QuantEcon/quantecon-book-theme
```

to use the `latest` master branch for the `quantecon-book-theme` bleeding edge until `0.2.1` is released in PyPI

**Comment by aadsouza** [comment]
@mmcky should i be able to test the launch button from the pr itself or will I need to `publish` first? the latest netlify preview doesn't resolve #128

**Comment by mmcky** [comment]
> @mmcky should i be able to test the launch button from the pr itself or will I need to `publish` first? the latest netlify preview doesn't resolve #128

Just use `netlify` previews

File: lectures/_config.yml [hunk 1]
```diff
@@ -23,6 +23,7 @@ sphinx:
       repository_url: https://github.com/QuantEcon/lecture-datascience.myst
       repository_branch: main
       nb_repository_url: https://github.com/QuantEcon/lecture-datascience.notebooks
+      nb_repository_branch: main
```
Comment (mmcky) [inline lectures/_config.yml:26]: @AakashGfude is this the correct way to specify `main` for the notebook launcher in `_config.yml`?

**Comment by mmcky** [comment]
@aadsouza let me know how this goes. We will need to document these things in the theme. But we rushed some things out a bit to get `datascience` previews up to date.

**Comment by aadsouza** [comment]
@mmcky so that did not work
an observation here https://github.com/QuantEcon/quantecon-book-theme/blob/master/quantecon_book_theme/launch.py 

we have 
```
    # If so, insert the URLs depending on the configuration
    config_theme = app.config["html_theme_options"]
    launch_buttons = config_theme.get("launch_buttons", {})

    if config_theme.get("nb_repository_url"):
        repo_url = _get_repo_url(config_theme)

        # Parse the repo parts from the URL
        org, repo, repo_subpath = _split_repo_url(repo_url)
        if repo_subpath:
            repo_url = repo_url.replace("/" + repo_subpath, "")
            repo_subpath += (
                "/"  # compatibility of code for cases which dont have this var
            )

        if org is None and repo is None:
            # Skip the rest because the repo_url isn't right
            return

        branch = _get_branch(config_theme)
```        
and
```
def _get_branch(config_theme):
    branch = config_theme.get("repository_branch")
    if not branch:
        branch = "master"
    return branch
```
throughout the file there is no `"nb_branch"`.

**Comment by aadsouza** [comment]
so `nb_branch` appears to be redundant and if i understand this correctly, it probably should use `repository_branch` which is `main`

**Comment by mmcky** [comment]
@AakashGfude will clarify what needs to be set. I will update this to branch with the latest `main` to ensure you're using the right version of the theme.

File: lectures/environment.yml [hunk 2]
```diff
@@ -13,7 +13,7 @@ dependencies:
     - fiona 
     - geopandas 
     - pyLDAvis >= 3.3.0
-    - git+https://github.com/QuantEcon/quantecon-book-theme/tree/notebook/jupyterlab
+    - git+https://github.com/QuantEcon/quantecon-book-theme
```
Comment (mmcky) [inline lectures/environment.yml:16]: @AakashGfude do you know why github sometime shows an incorrect diff?

Comment (aadsouza) [reply lectures/environment.yml:16]: @mmcky i had taken that stuff out following this https://github.com/QuantEcon/lecture-datascience.myst/pull/130#issuecomment-852665407

**Comment by mmcky** [comment]
@aadsouza I am going to close this PR as it has a messy commit history anyway for a single line change in `_config.yml` and I will check with @AakashGfude on the correct syntax to ensure `main` is used and not `master`

File: lectures/environment.yml (hunk 2 above)
Comment (mmcky) [reply lectures/environment.yml:16]: @aadsouza the latest master has pinned versions for `quantecon-book-theme` though

**Pull request closed without merging**